From: Ismael Luceno <ismael@iodev.co.uk>
Date: Wed, 3 Dec 2014 16:41:29 -0300
Subject: [mawk PATCH] Fix implementation of nextfile statement

Mawk behaves incorrectly when using the nextfile statement. It marks the
file as dead, but does not check such state and thus ends reading from
a closed fd.

The problem is fixed by simulating an EOF.

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 fin.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fin.c
+++ b/fin.c
@@ -189,6 +189,7 @@
 		close(fin->fd);
 	}
 
+	fin->flags |= EOF_FLAG;
 	fin->limit =
 	    fin->buff =
 	    fin->buffp = &dead;	/* marks it semi_closed */
